lib/VariableStore.js

```diff
@@ -31,17 +32,19 @@ function VariableStore( vars ){
   vs.var('foo').set('bar');
 **/
 VariableStore.prototype.var = function( key, val ){
-  if( !check.nonEmptyString( key ) ){
+  if( _.isEmpty( key ) || !_.isString( key) )  {
```

Comment:
I don't believe this is correct, my understanding of `_.isEmpty` is that it's for collections, not strings.

You might have to write it `isString(x) && !!x.length`

Reply:
`_.isEmpty` ends up checking `.length`, so it does work: https://github.com/lodash/lodash/blob/4.17.14/lodash.js#L11483-L11487

I agree that reading through [the docs](https://lodash.com/docs/4.17.14#isEmpty), it sounds like it shouldn't work, unless the definition of a "collection" is anything that has a `length` property.